### PR TITLE
[pkg] fix  deferring unsafe method "Close" on type "*os.File"

### PR DESCRIPTION
**Pull request by huof6829**
# Description

fix gosec error: [Deferring unsafe method "Close" on type "*os.File"](https://github.com/iotexproject/iotex-core/security/code-scanning/68)

Fixes   https://github.com/iotexproject/iotex-core/issues/3532

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/doc/doc.go [hunk 1]
```diff
@@ -32,19 +32,17 @@ func GenMarkdownTreeCustom(c *cobra.Command, dir string, name string, path strin
 		filename = filepath.Join(path, "README.md")
 	}
 
-	f, err := os.Create(filename)
+	f, err := os.Create(filepath.Clean(filename))
 	if err != nil {
 		return err
 	}
-	defer f.Close()
```
Comment (dustinxie) [inline ioctl/doc/doc.go:39]: can you check why security scan considers `f.Close()` as "unsafe method"? does it mean the error is not checked? like you need to:
```
defer func() {
    return f.Close()
}
```

Comment (huof6829) [reply ioctl/doc/doc.go:39]: error is skipped by `defer f.Close()`. it's unsafe.  error should return. 
`defer func() error {
    return f.Close()
}` 
 or  `return f.Close()` at the end of processing is both ok

Comment (dustinxie) [reply ioctl/doc/doc.go:39]: i see, so should check/catch the error inside `defer`

if `return f.Close()` at the end, also need to handle `f.Close()` in L42 and L45, so defer is better

Comment (Liuhaai) [reply ioctl/doc/doc.go:39]: current code is acceptable, you could find a new way to improve it

Comment (huof6829) [reply ioctl/doc/doc.go:39]: ok

File: pkg/recovery/recovery.go [hunk 2]
```diff
@@ -79,14 +79,18 @@ func LogCrash(r interface{}) {
...
 		return
 	}
-	defer f.Close()
 	if err := pprof.WriteHeapProfile(f); err != nil {
 		log.S().Errorf("crashlog: write heap profile error: %v", err)
+		return
+	}
+	if err = f.Close(); err != nil {
+		log.S().Errorf("crashlog: close heap profile error: %v", err)
+		return
```
Comment (dustinxie) [inline pkg/recovery/recovery.go:93]: also move to defer

File: ioctl/doc/doc.go [hunk 3]
```diff
@@ -36,8 +36,9 @@ func GenMarkdownTreeCustom(c *cobra.Command, dir string, name string, path strin
...
-
+	defer func() error {
+		return f.Close()
```
[the remark this answers is not in the document]
Comment (huof6829) [reply ioctl/doc/doc.go]: modified

File: pkg/recovery/recovery.go (hunk 2 above)
Comment (huof6829) [reply pkg/recovery/recovery.go]: done

**Review by Liuhaai — APPROVED** [review]

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**